cumulus/pallets/xcmp-queue/src/migration.rs

```diff
@@ -247,4 +347,64 @@ mod tests {
 			assert_eq!(v1.xcmp_max_individual_weight, v2.xcmp_max_individual_weight.ref_time());
 		});
 	}
+
+	#[test]
+	#[allow(deprecated)]
+	fn test_migration_to_v4() {
+		new_test_ext().execute_with(|| {
+			let storage_version = StorageVersion::new(3);
+			storage_version.put::<Pallet<Test>>();
+
+			let v2 = v2::QueueConfigData {
+				drop_threshold: 5,
+				suspend_threshold: 2,
+				resume_threshold: 1,
+				..Default::default()
+			};
+
+			frame_support::storage::unhashed::put_raw(
+				&crate::QueueConfig::<Test>::hashed_key(),
+				&v2.encode(),
+			);
+
+			v4::MigrationToV4::<Test>::on_runtime_upgrade();
```

Comment:
You can also call the pre and post hooks.

Reply:
Done